--- run-sepp.sh	2017-07-05 17:30:56.000000000 -0700
+++ run-sepp.sh.stefan	2017-09-25 15:27:46.723558612 -0700
@@ -1,13 +1,45 @@
 #!/bin/bash  
 
 if [ $# -lt 2 ]; then
-   echo USAGE: $0 "[input fragments file] [output prefix] [optional: -x number-of-cores ] [optional: -A alignment subset size] [optional: -P placement subset size] [optional: any other SEPP argument]
-   Optional commands need not be in order. Any SEPP option can also be passed. For example, use
-   -x 8
-   to make SEPP us 8 threads"
+	echo "This is SEPP for GreenGenes 13.8
+USAGE: $0 [deblur otu table] [output prefix] <options>
+
+input fragments file:
+  An deblur OTU table in biom format. Its features (V4 
+  amplicon sequences) will be used as fragments.
+output prefix:
+  an arbitrary name that identifies this run.
+
+Optional arguments:
+  -f
+    If the flag -f is raised, the input is expected to be a
+    fasta file (one line per sequence) of the fragments to be
+    inserted into the reference phylogeny instead of providing
+    an OTU table in biom format.
+  -x
+    Number of CPU cores to be used.
+  -A
+    Alignment subset size.
+  -P
+    Placement subset size.
+  -t
+    Filename for the reference pyholgeny in Newick format.
+    Default is GreenGenes 13.8.
+  -a
+    Filename for the reference alignment.
+    Should be in sync with the reference phylogeny in PyNAST format.
+    Default is GreenGenes 13.8.
+Optional commands need not be in order. Any SEPP option can also
+be passed. For example, use '-x 8' to make SEPP us 8 threads.
+"
    exit 1;
 fi
 
+echo "INFO: SEPP uses a lot of memory when merging found placements into a tree.
+Consider running SEPP in a grid environment instead of your local laptop!
+Next time, try this command:
+echo 'seppgg $1 $2 -x 8' | qsub -d $PWD -V -l walltime=12:00:00,nodes=1:ppn=8,pmem=8gb -N seppgg.$2"
+
 tmpbase=`mktemp -d -t 'tmp.XXXXXXXXXX'`
 if [ $? -ne 0 ]; 
 then
@@ -49,13 +81,16 @@
 name=$1
 shift
 
+# input type, default is a biom table
+inputtype='biom' 
+
 # SEPP placement and alignment subset sizes
 p=5000
 a=1000
 
 opts=""
 
-while [[ $# -gt 1 ]]
+while [[ $# -gt 0 ]]
 do
 	key="$1"
 
@@ -68,6 +103,18 @@
 			p="$2"
 			shift # past argument
 			;;
+		-t|--referenceTree)
+			t="$2"
+			shift # past argument
+			;;
+		-a|--referenceAlignment)
+			alg="$2"
+			shift # past argument
+			;;
+		-f|--fastaInput)
+			inputtype="fasta"
+			#~ shift
+			;;
 		*)
 			opts="$opts"" ""$key"" ""$2"
 			shift # past argument
@@ -79,32 +126,50 @@
 
 DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 
-# Reference tree
-t="$DIR/ref/reference-gg-raxml-bl-rooted-relabelled.tre"
-# Reference alignment
-alg="$DIR/ref/gg_13_5_ssu_align_99_pfiltered.fasta"
+if [ -z $t ]; then
+	# resort to the default reference tree, if not specified via command line argument
+	t="/opt/anaconda1anaconda2anaconda3/share/seppgg/reference-gg-raxml-bl-rooted-relabelled.tre"
+fi;
+if [ -z $alg ]; then
+	# resort to the default reference alignment, if not specified via command line argument
+	alg="/opt/anaconda1anaconda2anaconda3/share/seppgg/gg_13_5_ssu_align_99_pfiltered.fasta"
+fi;
 # RAxML info file generated when creating the reference RAxML tree
-rxi="$DIR/ref/RAxML_info-reference-gg-raxml-bl.info"
+rxi="/opt/anaconda1anaconda2anaconda3/share/seppgg/RAxML_info-reference-gg-raxml-bl.info"
 
 set -e
 
-python $DIR/sepp/run_sepp.py -P $p -A $a -t $t -a $alg -r $rxi -f $f -cp $tmpssd/chpoint-$name -o $name $opts -d $tmp/ -p $tmpssd 1>sepp-$name-out.log 2>sepp-$name-err.log
+if [[ $inputtype == 'fasta' ]]; then
+	echo "INFO: using fasta input file.";
+else
+	echo "INFO: extracting fragments from BIOM table."
+	filenameFragments=`basename $f`.fragments;
+	fragments=`biom summarize-table --observations -i $f | tail -n+16 | cut -d ":" -f 1 | tr [A-Z] [a-z] | sort -u`;
+	touch $filenameFragments;
+	for frg in $fragments; do
+		echo ">$frg" >> $filenameFragments;
+		echo "$frg" >> $filenameFragments;
+	done;
+	f=$filenameFragments
+fi;
+
+python /opt/anaconda1anaconda2anaconda3/share/seppgg/sepp/run_sepp.py -P $p -A $a -t $t -a $alg -r $rxi -f $f -cp $tmpssd/chpoint-$name -o $name $opts -d $tmp/ -p $tmpssd 1>$name-sepp-out.log 2>$name-sepp-status.log
 
-tail sepp-$name-*
+tail $name-sepp-*
 
 cp -r $tmpssd $tmp;
 
 cp $tmp/${name}_placement.json .
 cp $tmp/${name}_rename-json.py .
 
-gbin=$( dirname `grep -A1 "pplacer" $DIR/sepp/.sepp/main.config |grep path|sed -e "s/^path=//g"` )
-
-$gbin/guppy tog ${name}_placement.json
+/opt/anaconda1anaconda2anaconda3/bin/guppy tog ${name}_placement.json
 
 cat ${name}_placement.tog.tre | python ${name}_rename-json.py > ${name}_placement.tog.relabelled.tre
 
-$gbin/guppy tog --xml ${name}_placement.json
+/opt/anaconda1anaconda2anaconda3/bin/guppy tog --xml ${name}_placement.json
 
 cat ${name}_placement.tog.xml | python ${name}_rename-json.py > ${name}_placement.tog.relabelled.xml
 
 echo output files are at ${name}_placement.* and more files are at $tmp . Consider removing $tmp if its files are not needed. 
+
+echo "INFO: run was successful. You are most likely only interested in the file ${name}_placement.relabelled.tre."
